com.unity.perception/Runtime/GroundTruth/PerceptionCamera.cs

```diff
@@ -312,12 +313,25 @@ void OnValidate()
                 m_Labelers = new List<CameraLabeler>();
         }
 
+        // Convert the Unity 4x4 projection matrix to a 3x3 matrix
+        // ReSharper disable once InconsistentNaming
+        static float3x3 ToProjectionMatrix3x3(Matrix4x4 inMatrix)
```

Comment:
An alternative to this is taking the outMatrix = transpose(inverse(inMatrix)) 

Less performant, but I think more correct in terms of maintaining the dimensionality of the transform.  Also will correctly handle scaling in case det(matrix) != 1  (a bit of hand waiving here, would confirm this before actually doing it!)

Reply:
So, my main issue that I have here, is that the dataset insights code just wants to use a 3x3 matrix. From what I could make sense of, it looks like they were just using it for rotation (and I guess scaling) but the position was ignored. Also, I don't think that they had any use for sheering. Anyway, I'm not really sure what I need to do to go from a 4x4 to a 3x3 but it appeared that in a 4x4 homogenous matrix, the upper 3x3 correlate directly with a 3x3 affine transformation. But I really don't know

Reply:
I would argue that they should be doing it the correct linear algebra way too in that case :) 

Stripping the matrix down doesn't compensate for scale properly since it's possible for mat_4_4 to be non 1 for example and this would then in effect scale the rest of the matrix down, which is why you want to do that transpose-inverse thing. 

One idea is to just maintain it as 4x4 Unity side all the way through and then export it as 3x3, although I'd argue better to export as 4x4 and let dataset insights convert it. 

This could also be chalked up to convention as well, so feel free to ignore all of this as well - functionally it really doesn't make much of a difference